### PR TITLE
The Hive playbooks

### PR DESCRIPTION
**Pull request by vu-socprime**
Added The Hive Custom API Logic App Connector and 3 playbooks

### CODE REVIEW

File: Solutions/TheHive/Playbooks/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "https://github.com/socprime/Azure-Sentinel/raw/PS-1681_TheHive_Playbooks/Solutions/TheHive/Playbooks/TheHive-CreateCase/azuredeploy.json"
+                },
+                "parameters": {
+                    "PlaybookName": {
+                        "Value": "[parameters('TheHive-CreateCase')]"
+                    },
+                    "onPremiseGatewayName": {
+                        "Value": "[parameters('onPremiseGatewayName')]"
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/TheHive/Playbooks/azuredeploy.json:120]: On Premise Data gateway should be deployed under the same subscription and resource group as playbooks. Nevertheless it is not necessary to provide valid onPremiseGateway name during the deployment. Playbooks will be deployed in any case. Link to the gateway can be changed later during configuring API connections.

**Comment by v-spadarthi** [comment]
@rushriva : Please provide your comments

**Comment by v-spadarthi** [comment]
@rushriva : Could you please review the playbooks. Thanks!

File: Solutions/TheHive/Playbooks/azuredeploy.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ospeka) [reply Solutions/TheHive/Playbooks/azuredeploy.json:120]: You need to have valid On-Premises Data Gateway under the same subscription and resource group for deploy. The gateway must at least exist for deploy. Please, try again and if it fails ping me or vu-socprime.

**Review by anki-narravula — CHANGES_REQUESTED** [review]
Hi @vu-socprime - Can you please make following changes. Thanks in advance

1. For all playbooks, metadata object is missing. With the new template spec (for gallery), this is mandatory. Please look at the [ latest guide for playbooks contribution](https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide)
2. For all playbooks custom connector name must be included in the Parameters and use the parameter directly in "id" attribute of API. For example 
```
 "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', variables('customApis_vendorproduct_name'))]"
 "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', parameter('customApis_rapid7'))]"
```
3. In Custom Connector - name of the custom connector must be a parameter and keep default value (same as the current) also we suggest making host url also as parameter
4. Wherever sentinel connection used, use Managed Identity always - [refer template for an example ](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json)
5. Please consider limited actions for Custom connector actions (whatever relevant or closer to SOAR perspective). Also please describe in readme (if it is missing)
6. (If not implemented) Make sure playbook should not fail in case of API does not find entity / specified identity. Rather it should write in Sentinel comment that "Unable to find ......., hence no action taken) 
7. Make sure images are loading in readme files

**Comment by vu-socprime** [comment]
> Hi @vu-socprime - Can you please make following changes. Thanks in advance
> 
> 1. For all playbooks, metadata object is missing. With the new template spec (for gallery), this is mandatory. Please look at the [ latest guide for playbooks contribution](https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide)
> 2. For all playbooks custom connector name must be included in the Parameters and use the parameter directly in "id" attribute of API. For example
> 
> ```
>  "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', variables('customApis_vendorproduct_name'))]"
>  "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', parameter('customApis_rapid7'))]"
> ```
> 
>  3. In Custom Connector - name of the custom connector must be a parameter and keep default value (same as the current) also we suggest making host url also as parameter
> 4. Wherever sentinel connection used, use Managed Identity always - [refer template for an example ](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json)
> 5. Please consider limited actions for Custom connector actions (whatever relevant or closer to SOAR perspective). Also please describe in readme (if it is missing)
> 6. (If not implemented) Make sure playbook should not fail in case of API does not find entity / specified identity. Rather it should write in Sentinel comment that "Unable to find ......., hence no action taken)
> 7. Make sure images are loading in readme files

Hi @anki-narravula 
The playbooks were updated according to the feedback

**Review by anki-narravula — CHANGES_REQUESTED** [review]
Hi @vu-socprime  - We are good now. Can you please change the azure deploy links to point to master now. Also please see the readme file of TheHive-CreateCase, screenshot is not loading

**Comment by vu-socprime** [comment]
> Hi @vu-socprime - We are good now. Can you please change the azure deploy links to point to master now. Also please see the readme file of TheHive-CreateCase, screenshot is not loading

Hi @anki-narravula - Screenshot loading fixed. Links already point to master.

**Comment by anki-narravula** [comment]
> > Hi @vu-socprime - We are good now. Can you please change the azure deploy links to point to master now. Also please see the readme file of TheHive-CreateCase, screenshot is not loading
> 
> Hi @anki-narravula - Screenshot loading fixed. Links already point to master.

Hi @vu-socprime  - I am somehow seeing that it is referencing to socprime branch. Below is the screenshot attached
image

**Comment by vu-socprime** [comment]
> > > Hi @vu-socprime - We are good now. Can you please change the azure deploy links to point to master now. Also please see the readme file of TheHive-CreateCase, screenshot is not loading
> > 
> > 
> > Hi @anki-narravula - Screenshot loading fixed. Links already point to master.
> 
> Hi @vu-socprime - I am somehow seeing that it is referencing to socprime branch. Below is the screenshot attached image

@anki-narravula - Fixed

**Review by anki-narravula — APPROVED** [review]
Looks good. Approved.

**Pull request merged**